doc/boot-partition.md

```diff
@@ -130,21 +134,34 @@ i.e. valid for all architectures (s390, see below)
 - order: /boot, swap, /
 - ext4
 
-x86-legacy
-----------
-- grub boot.img (stage 1) in MBR
-- install core.img (stage 1.5) in free space between partition table and 1. partition
-- if there isn't enough free space -> create additional partition
-- install all grub parts on same disk
-
-x86-efi
--------
-if system efi partition is already there -> reuse it
-if not create:
-- /boot/efi
-- fat32
-- size: 33 MB, 200MB
-- limit: below 2TB
+## x86-legacy
+
+*[this assumes grub will be installed into mbr]*
+
+- required size for grub embedding (currently): 84 kB; to be safe, check for >= 256 kB
+- if dos partition table
+    - embed in unused space before 1st partition
+    - fail if not enough space
+    - note: grub-install may later fail if it detects the area is used by some other weird software
+- if gpt partition table
+    - BIOS boot partition must exist (gpt type 21686148-6449-6e6f-744e-656564454649); parted: set flag `bios_grub` to `on`
+    - if it doesn't exist, create:
+        - size: [1MB, 2MB, 100MB]
+        - no fs
+        - no mount point
+  - if existing BIOS boot partition is too small, delete and re-create
+  - grub will use 1st BIOS boot partition it finds, co-op with other grub instances on same disk not possible
+- note: `yast bootloader` will be responsible for boot flag handling, if necessary
+  
```

Comment:
I'd like to add that.
1. Don't use gpt-sync-mbr or hybrid mbr. Only standard "protective mbr".
2. Make sure the boot flag is set in protective mbr and also in normal mbr, because some bios is stubborn to fix themselves and insist to detect the active/boot flag before handing over control to mbr boot code. It seems to me YaST has fixed this already, but still can be reproduced in our hybrid iso images.